web/src/unread_ops.ts

```diff
@@ -52,13 +56,190 @@ export function confirm_mark_all_as_read(): void {
     const html_body = render_confirm_mark_all_as_read();
 
     const modal_id = confirm_dialog.launch({
-        html_heading: $t_html({defaultMessage: "Mark all messages as read?"}),
+        html_heading: $t_html({defaultMessage: "Choose messages to mark as read"}),
         html_body,
         on_click() {
-            mark_all_as_read(modal_id);
+            mark_selected_options_as_read(modal_id);
         },
         loading_spinner: true,
     });
+
+    const default_message_count = countMessages("topics_not_followed");
+
+    $("#message_count").text(
+        $t(
+            {
+                defaultMessage:
+                    "{count, plural, one {# message} other {# messages}} will be marked as read.",
+            },
+            {count: default_message_count},
+        ),
+    );
+
+    $("#mark_as_read_option").on("change", function () {
+        const selected_option = String($(this).val());
+        const messageCount = countMessages(selected_option);
+
+        $("#message_count").text(
+            $t(
+                {
+                    defaultMessage:
+                        "{count, plural, one {# message} other {# messages}} will be marked as read.",
+                },
+                {count: messageCount},
+            ),
+        );
+    });
+}
+
+function countMessages(selected_option: string): number {
+    let count = 0;
+
+    switch (selected_option) {
+        case "all_messages": {
+            count = getAllMessagesCount();
```

Comment:
Yes @amanagr , I have also done the same thing. But as you pointed it out, I have wrote redundant code, as I could directly call the function instead of defining another function that call the `unread.get_unread_message_count()`.

Corrected!